### ISSUE TITLE
Issues Identified with GitLab Trigger Node

### ISSUE THREAD

**Issue by chodorenko** [n8n-io/n8n#12054]
### Bug Description

We have identified several issues with the GitLab Trigger node in n8n, one of which poses a security concern:

1. When creating this object in n8n, it is not possible to set the Secret Token or enable SSL Verification. This is a significant issue because the absence of token verification could lead to unauthorized processing and potential system compromise. Additionally, the lack of SSL verification allows for the possibility of host spoofing for the webhook endpoint.

2. Since the webhook is automatically recreated in GitLab, it cannot be edited or have its key verified on the n8n side. A similar issue has been reported here: https://github.com/n8n-io/n8n/issues/10621.

3. The GitLab Trigger editing form in n8n does not provide a notification that a webhook will be automatically created on the GitLab server. This often causes confusion when users manually create a webhook in GitLab, only to find another webhook with the same parameters appearing unexpectedly.

We hope these issues can be addressed to improve the functionality and security of the GitLab Trigger node.

### To Reproduce

1. Create GitLab Trigger Node in n8n
2. Monitor webhooks on a given project through Settings > Webhooks

### Expected behavior

1. Add Secret Token and Enable SSL Verification parameters to the GitLab Trigger Node.
2. Automatically validate the Secret Token on the n8n side.
3. Add a warning or note stating: "The webhook is created automatically when applying or modifying parameters."

### Operating System

Debian 12

### n8n Version

1.69.2

### Node.js Version

-

### Database

SQLite (default)

### Execution mode

main (default)

**Comment by Joffcom** [comment]
Hey @chodorenko,

We have created an internal ticket to look into this which we will be tracking as "GHC-561"

**Comment by Joffcom** [comment]
Hey @chodorenko 

Thanks for creating this issue, looking at the 3 things you have reported the only one that looks to be a bug is number 2 which is a duplicate of another issue.

Issue 3 I have not heard any reports of users manually making a webhook for the Gitlab node but I will make sure we update our documentation to mention this.

This leaves issue 1 which is not a bug but more of an enhancement / feature request for the node and one I fully agree we should add although if we did add it the secret would be internally managed like it is for a couple of other nodes that support this type of verification.

Based on all of the above I am going to close this issue as we do only use GitHub issues to track bugs and this is 1 documentation update, 1 duplicate issue and an enhancement. This however doesn't mean we won't address any of this and I have added number 1 to my task list as I really do think it makes sense to add this sooner rather than later.

Let me know if you have any questions on this.
